app/models/neighbour_response.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+class NeighbourResponse
+  include ActiveModel::Model
+
+  STAGES = %w[about_you thoughts response check].freeze
+
+  TAGS = %i[design use privacy light access noise traffic other].freeze
+  RESPONSE_PARAMS  = [:name, :email, :address, :response, :summary_tag, TAGS, :tags, { tags: [] }].freeze
+  PERMITTED_PARAMS = [:stage, :move_next, :move_back, :planning_application_id,
+                      { neighbour_response: RESPONSE_PARAMS }].freeze
+
+  attr_reader :params, :errors
+
+  def initialize(params)
+    @params = params.permit(*PERMITTED_PARAMS)
+    @errors = ActiveModel::Errors.new(self)
```

Comment:
Is there a reason why we're not using the standard Active Model setup? Including `ActiveModel::Model` sets up errors and has helpers for assigning attributes from a hash. If you include the `ActiveModel::Attributes` module you can use the `attribute` DSL to define the attributes and then use standard validation patterns, e.g.

``` ruby
class NeighbourResponse
  include ActiveModel::API
  include ActiveModel::Attributes

  attribute :name, :string
  attribute :email, :string

  validates :name, presence: true, if: -> { stage == "about_you" }
end
```

I'd also move the permitted params stuff to the more conventional controller location.